City-Bureau/city-scrapers-indianapolis
# Fix scrapy validate bug

## What's this PR do?
Fixes a mysterious error where city-scraper-core's custom command, `scrapy validate <spider-name>`, was failing locally and in CI. It updates and modifies a few of our dependencies and sets python 3.9 as a requirement in the Pipfile.

## Why are we doing this? 
`scrapy validate` is an important part of our CI process and we need it to work as expected. This caused problems for a recent PR: #14 

The error looked like this:
```
Traceback (most recent call last):
  File "/home/runner/work/city-scrapers-indianapolis/city-scrapers-indianapolis/.venv/bin/scrapy", line 8, in <module>
    sys.exit(execute())
  File "/home/runner/work/city-scrapers-indianapolis/city-scrapers-indianapolis/.venv/lib/python3.9/site-packages/scrapy/cmdline.py", line 153, in execute
    cmd.add_options(parser)
  File "/home/runner/work/city-scrapers-indianapolis/city-scrapers-indianapolis/.venv/lib/python3.9/site-packages/city_scrapers_core/commands/validate.py", line 23, in add_options
    parser.add_option(
AttributeError: 'ScrapyArgumentParser' object has no attribute 'add_option'
```
It's a bit mysterious to me why exactly we were getting this error, but my best guess is it related to some kind of dependency conflict. Many of our core packages (particularly scrapy-sentry and city-scrapers-core) have not been touched in years. See [this PR](https://github.com/City-Bureau/city-scrapers-cle/pull/67) in city-scrapers-cle or [this PR](https://github.com/City-Bureau/city-scrapers/pull/1056) for more background on these issues.

## Steps to manually test
1) If you have an existing pipenv virtual env for this project, delete it:
```
pipenv --rm
```
2) Reinstall the project
```
pipenv sync --dev
```
3) Make sure `scrapy validate` works as expected and doesn't throw any errors:
```
pipenv run scrapy validate --all
```

## Are there any smells or added technical debt to note? 
- I wish I knew exactly why we were getting this bug, but this combo of dependencies appears to solve it.
- I don't love that we are resolving our city-scrapers-core dependency from our Github repo instead of PyPi, ie this change:
```
city-scrapers-core = {ref = "main", git = "https://github.com/City-Bureau/city-scrapers-core.git", extras = ["azure"]}
```
I experimented with excluding this change but it _seemed_ to help resolve the problem. `city-scrapers-core` is overdue for upgrades and that might help resolve this issue.
